### PR TITLE
Functional test for source VM that wouldn't go down

### PR DESCRIPTION
**Pull request by jakub-dzon**
```release-notes
NONE
```
Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

**Review by pkliczewski — APPROVED** [review]

File: tests/ovirt/basic_vm_import_negative_test.go [hunk 1]
```diff
@@ -134,9 +138,44 @@ var _ = Describe("VM import", func() {
...
+
+		Consistently(func() (*v2vv1alpha1.VirtualMachineImportCondition, error) {
+			retrieved, err := f.VMImportClient.V2vV1alpha1().VirtualMachineImports(f.Namespace.Name).Get(created.Name, metav1.GetOptions{})
+			if err != nil {
+				return nil, err
+			}
+			condition := conditions.FindConditionOfType(retrieved.Status.Conditions, v2vv1alpha1.Processing)
+			return condition, nil
+
+		}, 5*time.Minute, time.Minute).Should(BeNil())
```
Comment (machacekondra) [inline tests/ovirt/basic_vm_import_negative_test.go:174]: Should we also check that after the timeout, we get the error?

**Review by machacekondra — APPROVED** [review]

File: tests/ovirt/basic_vm_import_negative_test.go (hunk 1 above)
Comment (jakub-dzon) [reply tests/ovirt/basic_vm_import_negative_test.go:174]: What error do you have in mind? The timeout here only specifies how long the assertion should be kept, it does not return error after the timeout. In this case it's 5 minutes, which is as long as the ovirt client waits for the VM to be down in one reconciliation and that long I wanted to make sure that it doesn't start processing.

Comment (machacekondra) [reply tests/ovirt/basic_vm_import_negative_test.go:174]: I mean this one: https://github.com/kubevirt/vm-import-operator/blob/master/pkg/providers/ovirt/client/ovirt-client.go#L139

When the timeout (5min) is out, the stopVm method should return error.

Comment (jakub-dzon) [reply tests/ovirt/basic_vm_import_negative_test.go:174]: It does but on high level it is translated only to reconciliation error and CR re-queuing (https://github.com/kubevirt/vm-import-operator/blob/12de4cca49c113ce028fcb84934a64541009922e/pkg/controller/virtualmachineimport/virtualmachineimport_controller.go#L239); nothing we can detect except for verifying it doesn't progress beyond mapping rules verification.

**Pull request merged**